habitat/datasets/rearrange/samplers.py

```diff
@@ -339,9 +343,17 @@ def sample(
             and num_pairing_tries < self.max_sample_attempts
         ):
             num_pairing_tries += 1
-            new_object = self.single_sample(sim, snap_down, vdb)
+            if len(new_objects) == 0:
+                # no objects sampled yet
+                new_object, receptacle = self.single_sample(
+                    sim, snap_down, vdb, target_receptacle
+                )
+            else:
+                new_object, receptacle = self.single_sample(
+                    sim, snap_down, vdb
+                )
             if new_object is not None:
-                new_objects.append(new_object)
+                new_objects.append((new_object, receptacle))
```

Comment:
So we manually sample the 1st object from the specified target receptacle and then sample others randomly? Seems a bit contrived and is not documented.